storage/remote/otlptranslator/prometheusremotewrite/metrics_to_prw_test.go

```diff
@@ -235,6 +237,356 @@ func TestFromMetrics(t *testing.T) {
+			convertToNHCB: false,
+			inputSeries: []pmetric.Metric{
+				createOtelExpHistogram("test_histogram_1", pmetric.AggregationTemporalityDelta, ts),
+				createOtelExplicitHistogram("test_histogram_2", pmetric.AggregationTemporalityCumulative, ts),
+			},
+			expectedSeries: []prompb.TimeSeries{
+				createPromNativeHistogram("test_histogram_1", prompb.Histogram_GAUGE, ts),
+				{
+					Labels: []prompb.Label{
+						{Name: "__name__", Value: "test_histogram_2_bucket"},
+						{Name: "le", Value: "1"},
+						{Name: "test_label", Value: "test_value"},
+					},
+					Samples: []prompb.Sample{{Value: 10, Timestamp: 100000}},
+				},
+				{
+					Labels: []prompb.Label{
+						{Name: "__name__", Value: "test_histogram_2_bucket"},
+						{Name: "le", Value: "2"},
+						{Name: "test_label", Value: "test_value"},
+					},
+					Samples: []prompb.Sample{{Value: 20, Timestamp: 100000}},
+				},
+				{
+					Labels: []prompb.Label{
+						{Name: "__name__", Value: "test_histogram_2_bucket"},
+						{Name: "le", Value: "+Inf"},
+						{Name: "test_label", Value: "test_value"},
+					},
+					Samples: []prompb.Sample{{Value: 20, Timestamp: 100000}},
+				},
+				{
+					Labels: []prompb.Label{
+						{Name: "__name__", Value: "test_histogram_2_count"},
+						{Name: "test_label", Value: "test_value"},
+					},
+					Samples: []prompb.Sample{{Value: 20, Timestamp: 100000}},
+				},
+				{
+					Labels: []prompb.Label{
+						{Name: "__name__", Value: "test_histogram_2_sum"},
+						{Name: "test_label", Value: "test_value"},
+					},
+					Samples: []prompb.Sample{{Value: 30, Timestamp: 100000}},
+				},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			metrics := pmetric.NewMetrics()
+			rm := metrics.ResourceMetrics().AppendEmpty()
+			sm := rm.ScopeMetrics().AppendEmpty()
+
+			for _, s := range tc.inputSeries {
+				s.CopyTo(sm.Metrics().AppendEmpty())
+			}
+
+			c := NewPrometheusConverter()
+			settings := Settings{
+				AllowDeltaTemporality:   tc.allowDelta,
+				ConvertHistogramsToNHCB: tc.convertToNHCB,
+			}
+
+			_, err := c.FromMetrics(context.Background(), metrics, settings)
+
+			if tc.expectedError != "" {
+				require.EqualError(t, err, tc.expectedError)
+				return
+			}
+
+			require.NoError(t, err)
+			series := c.TimeSeries()
+
+			// Sort series to make the test deterministic.
+			testutil.RequireEqual(t, sortTimeSeries(tc.expectedSeries), sortTimeSeries(series))
+		})
+	}
+}
+
+func createOtelSum(name string, temporality pmetric.AggregationTemporality, ts time.Time) pmetric.Metric {
+	metrics := pmetric.NewMetricSlice()
+	m := metrics.AppendEmpty()
+	m.SetName(name)
+	sum := m.SetEmptySum()
+	sum.SetAggregationTemporality(temporality)
+	dp := sum.DataPoints().AppendEmpty()
+	dp.SetDoubleValue(5)
+	dp.SetTimestamp(pcommon.NewTimestampFromTime(ts))
+	dp.Attributes().PutStr("test_label", "test_value")
+	return m
+}
+
+func createPromCounter(name string, ts time.Time) prompb.TimeSeries {
```

Comment:
```suggestion
func createPromTimeseries(name string, ts time.Time) prompb.TimeSeries {
```

It's not really specifying the metric type here

Reply:
I've renamed this to `createPromFloatSeries` to distinguish between the float and histogram helper functions